src/main/java/flaxbeard/steamcraft/tile/TileEntitySteamCharger.java

```diff
@@ -91,87 +96,151 @@ public void updateEntity() {
-                            i++;
+                } else if (CrossMod.TINKERS_CONSTRUCT
+                  && this.getStackInSlot(0).getItem() instanceof ToolCore) {
+                    ToolCore item = (ToolCore) this.getStackInSlot(0).getItem();
+                    ItemStack stack = this.getStackInSlot(0).copy();
+                    NBTTagCompound tags = stack.getTagCompound();
+                    int damage = tags.getCompoundTag("InfiTool").getInteger("Damage");
+                    if (!tags.getCompoundTag("InfiTool").getBoolean("Broken")) {
+                        if (this.getSteamShare() > 0
+                            && tags.getCompoundTag("InfiTool").getInteger("Damage") > 0) {
+                            if (!this.isCharging) {
+                                // Steamcraft.log.debug("Charging");
+                                this.isCharging = true;
+                                worldObj.markBlockForUpdate(xCoord, yCoord, zCoord);
+                            }
+                        } else {
+                            if (this.isCharging) {
+                                // Steamcraft.log.debug("Not charging");
+                                this.isCharging = false;
+                                worldObj.markBlockForUpdate(xCoord, yCoord, zCoord);
+                            }
                         }
-                        float currentPerc = getChargingPercent(stack);
-                        if (prevPercent != currentPerc && Math.abs(prevPercent - currentPerc) > 0.01) {
-                            //log.debug("New percent: "+currentPerc);
-                            prevPercent = currentPerc;
-                            worldObj.markBlockForUpdate(xCoord, yCoord, zCoord);
+                        if (this.getSteamShare() > 8
+                            && tags.getCompoundTag("InfiTool").getInteger("Damage") > 0) {
+                            int i = 0;
+                            while (i < 4
+                                && (this.getSteamShare() > 8 && tags.getCompoundTag("InfiTool")
+                                    .getInteger("Damage") > 0)) {
+                                this.decrSteam(8);
+                                damage -= 1;
+                                tags.getCompoundTag("InfiTool").setInteger("Damage", damage);
+                                this.setInventorySlotContents(0, stack);
+                                i++;
+                            }
+                            float currentPerc =
+                                (float) (tags.getCompoundTag("InfiTool").getInteger(
+                                    "TotalDurability") - tags.getCompoundTag("InfiTool")
+                                    .getInteger("Damage"))
+                                    / (float) tags.getCompoundTag("InfiTool").getInteger(
+                                        "TotalDurability");
+                            ;
```

Comment:
Stray semicolons are just ignored in java. Which is kind of a pain since I didn't notice it.